fwdctl/src/dump.rs

```diff
@@ -21,6 +21,16 @@ pub struct Options {
         help = "Name of the database"
     )]
     pub db: String,
+
+    /// The key to start dumping from (if no key is provided, start from the beginning).
+    /// Defaults to None.
+    #[arg(
+        required = false,
+        value_name = "START_KEY",
+        default_value_t = String::from(""),
+        help = "Start dumping from this key (inclusive)."
+    )]
+    pub start_key: String,
```

Comment:
Use `Option<Box<[u8]>>`. We should describe the input format more specifically -- this should be a set of hex bytes which we should parse and produce a binary key.

This is pretty hard to get the parsing right, but it's possible to set the type here and implement a custom parser for the start_key, something like this:

```
fn key_parser(s: &str) -> Result<Box<[u8]>, ParseIntError> {
    let mut result = Vec::with_capacity(s.len() / 2);
    let mut iter = s.chars().peekable();
    while iter.peek().is_some() {
        let chunk = iter.by_ref().take(2).collect::<String>();
        result.push(u8::from_str_radix(&chunk, 16)?);
    }
    Ok(result.into_boxed_slice())
}
```

Then, you just set the attribute `value_parser = key_parser` in the `arg` section above and it works.

This code was only tested against some happy paths, and I see two small problems with it, but it's probably fine for this. The problems are:
 - If the string is an odd length, it "works" kinda; we should enforce that each slice is 2 characters
 - The error sucks because it doesn't tell you what part of the string was wrong, just that there was something wrong

It's also not super performant but I think that doesn't matter for this case.

Reply:
I think it makes more sense to have the start key string be converted into bytes the same way that we do in `fwdctl insert` so I did that. If you feel strongly we could do hex